packages/core/src/index.js

```diff
@@ -36,14 +36,28 @@ export const createParser = config => {
       const scale = get(props.theme, sx.scale, sx.defaults)
 
       if (typeof raw === 'object') {
-        cache.breakpoints =
-          cache.breakpoints ||
-          get(props.theme, 'breakpoints', defaults.breakpoints)
+        // If a child theme provides breakpoints, the cached ones should be updated
+        const shouldCacheBeBusted = get(props.theme, 'breakpoints', [])
```

Comment:
Thanks for looking into this! And yeah, the whole point of the caching was performance related so this looks like it negates a lot of that. This line in particular I think could be hoisted up above the `for` loop, and I suspect the conditional could be adjusted to minimize the impact. There should be a benchmark suite in the repo if you have time to run tests with different approaches. If the caching here doesn't make a huge impact, it might be something we can get rid of altogether

Reply:
Thanks for your feedback! I finally got around to make some changes to this PR to address your comments:

1. It now checks whether there's a key named `disableStyledSystemCache` in the provided `theme`. If not, is uses the cache just like before. If it is present, it takes the breakpoints from the provided theme (with a fallback to the defaults)
2. Concerning the benchmarks – I ran those with different implementations and couldn't really tell a significant difference in most cases. However, my results seemed to be a bit random anyway. The numbers differed quite a bit, on the same machine and without any significant background processes. If you have a way to run this more reliably, maybe we could make sure I didn't break anything performance-wise.